### PR TITLE
Replaces mocks with VCR calls

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Replaces mocked replies with server replies, stored as VCR cassettes.

Tests payment intent creation with a valid card and a insufficient funds card.

#### What should we test?

- green build

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/lib/stripe/payment_intent_validator_spec.rb [hunk 1]
```diff
@@ -38,14 +66,41 @@ module Stripe
...
 
         it "raises Stripe error with payment intent last_payment_error as message" do
+          expect {
+            Stripe::PaymentIntent.confirm(payment_intent.id)
+          }.to raise_error Stripe::StripeError, "Your card has insufficient funds."
+
           expect {
             validator.call
-          }.to raise_error Stripe::StripeError, "No money"
+          }.to raise_error Stripe::StripeError, "Your card has insufficient funds."
```
Comment (mkllnk) [inline spec/lib/stripe/payment_intent_validator_spec.rb:103]: Great! Now we can actually re-record this after any stripe updates and check if the specs still pass. This is huge!

File: spec/lib/stripe/payment_intent_validator_spec.rb [hunk 2]
```diff
@@ -5,30 +5,58 @@
...
 module Stripe
   describe PaymentIntentValidator do
-    describe "#call" do
-      let(:validator) { Stripe::PaymentIntentValidator.new(payment) }
-      let(:payment) { build(:payment, response_code: payment_intent_id) }
-      let(:payment_intent_id) { "pi_123" }
-      let(:stripe_account_id) { "abc123" }
-      let(:stripe_account_mock) { double(stripe_user_id: stripe_account_id) }
-      let(:payment_intent_response_mock) { { status: 200, body: payment_intent_response_body } }
+    let(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
```
Comment (mkllnk) [inline spec/lib/stripe/payment_intent_validator_spec.rb:8]: Can we keep a common test key in Bitwarden to make this easy?

**Review by mkllnk — APPROVED** [review]
Thank you! This is revolutionary.

**Review by dacook — APPROVED** [review]
Great to test the real-world response!

I found it a bit hard to review because there seemed to be multiple changes in one commit. But all good 👍 

I was thinking perhaps we can consider setting a record interval so that it automatically re-records (https://benoittgt.github.io/vcr/#/configuration/default_cassette_options?id=cassettes-can-set-their-own-options). But there's a few things to consider (do we do it on CI, and if so, how to store the updated cassette, and only do it on master.. and... maybe not)

**Comment by filipefurtad0** [comment]
Thanks both @mkllnk @dacook :raised_hands: 

> Can we keep a common test key in Bitwarden to make this easy?

I've now created a secret note with name `Stripe Secret Test API Key` on Bitwarden :+1: 

> I found it a bit hard to review because there seemed to be multiple changes in one commit. But all good 👍

Sorry for this! I got all excited to see the expected outcome on the second example, so I totally forgot to split this into two separate commits...

> I was thinking perhaps we can consider setting a record interval so that it automatically re-records

This is a great idea. I'll see if I can get that working.

PS: To me, PR #11701 served as an un-blocker: thank you so much @macanudo527 for initiating this work!!

**Pull request merged**